Topics/Tech_Stacks/build-android-app-using-github-actions.md

```diff
@@ -0,0 +1,158 @@
+your APK directly on GitHub.
+
+## Getting Started
+
+First, we will create a new Github Actions Workflow. A workflow is just a
+[YAML](https://yaml.org/) file that contains tiggers for when certain jobs
+(scripts) should run. 
+
+Create the file `.github/workflows/build-frontend.yml` to begin specifying the
+workflow.
+
+You should also add a name for the workflow at the top which will be visible in
+the GitHub Actions tab.
+
+```yml
+name: frontend-cd
+```
+
+![](./assets/actions-list.png)
+
+## Workflow Triggers 
+
+Triggers define when your jobs will run. We will setup the workflow to build
+whenever there is a push to main:
+
+```yml
+on:
+  push: 
+    paths: [ "Frontend/**" ]
+    branches: [ "main" ]
+```
+
+Furthermore, notive how I included the filter `paths: [ "Frontend/**" ]`, this
+is important because it avoid running our build unescesarily. For example, for
+my case, I have both frontend and backend code in the same repo. Thus, having
+this filter avoid building the frontend for backend pull requests.
+
+## Defining a Job
+
+### Setup Steps
+
+```yml
+jobs:
+  cd:
+    runs-on: ubuntu-latest
+
+    timeout-minutes: 20
+
+    steps:
+
+    # Clone the repo on the target machine (ubuntu-latest)
+    - name: Checkout code
+      uses: actions/checkout@v4
+
+    # Installs Node.js
+    - name: Set up Node.js
+      uses: actions/setup-node@v4
+
+    # Installs your prjects dependencies as defined in your package.json file
+    - name: Install dependencies
+      working-directory: Frontend/DebrisMine
+      run: npm install
+
+    # Install the Android SDK (for building the app)
+    - name: Setup Android SDK
+      uses: android-actions/setup-android@v2
+
+    # Install the Gradle build tool
+    - name: Setup Gradle
+      uses: gradle/gradle-build-action@v2
+```
+
+The schema here is fairly self explanatory. But for completeness, here we only define one job (named `cd`). It runs on an Ubuntu machine, and the job times out after 20 minutes (useful to avoid unexpected long builds). Notice how each step has a name, this is the name that will be visible when the action is run:
+
+![](./assets/action-steps.png)
+
+### Build steps
+
+```yml
+    # Optional pre-build step
+    # Make sure to cd to your android folder though!
+    - name: Prebuild Android
+      working-directory: Frontend/DebrisMine
+      run: |
+        npx expo prebuild -p android --no-install
+        cd android && chmod +x ./gradlew
+
+    # Building using gradle
+    - name: Build Android Release
+      working-directory: Frontend/DebrisMine
+      run: |
+        cd android && ./gradlew clean && ./gradlew app:assembleRelease
+```
+
+Since I use expo, I have a prebuild step before building which generates the android folder for me. If you are not using expo, then simply ommit this line `npx expo prebuild -p android --no-install`, and just cd into the android folder which you should have pushed to your repo.
```

Comment:
Maybe add a link to expo so the reader can explore it as well

Reply:
done